src/utils/hooks/useYoutubeAPI.js

```diff
@@ -0,0 +1,65 @@
+import { useCallback, useEffect, useState } from 'react';
+
+function useYoutubeAPI(search) {
+  const [scriptLoaded, setScriptLoaded] = useState(false);
+  const [clientInit, setClientInit] = useState(false);
+  const [responseData, setResponseData] = useState();
+
+  // Dynamically put the script to load GAPI and listen to when it's loaded
+  useEffect(() => {
+    if (document.getElementById('gapi-script')) {
+      return setScriptLoaded(true);
+    }
+    const scriptTag = document.createElement('script');
+    scriptTag.addEventListener('load', () => setScriptLoaded(true));
+    scriptTag.src = 'https://apis.google.com/js/client.js';
+    scriptTag.id = 'gapi-script';
+    document.body.appendChild(scriptTag);
+  }, []);
```

Comment:
I think this should be placed directly into your `public.html` file in the form of `<script></script>`, so that when you build the app this is already there. If not, you're prone to inject the script multiple times. Check the following image, on every render for the home page you're putting the client to load.

![Screen Shot 2021-04-13 at 10 33 43](https://user-images.githubusercontent.com/10038319/114579873-eb56ac80-9c43-11eb-9254-4a8188585148.png)